timelock-corruption-detection/src/main/java/com/palantir/history/sqlite/SqlitePaxosStateLogHistory.java

```diff
@@ -0,0 +1,98 @@
+import java.util.function.Function;
+
+import javax.sql.DataSource;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.mapper.immutables.JdbiImmutables;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.BindPojo;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+
+import com.palantir.history.mappers.AcceptorPaxosRoundMapper;
+import com.palantir.history.mappers.LearnerPaxosRoundMapper;
+import com.palantir.history.mappers.NamespaceAndUseCaseMapper;
+import com.palantir.history.models.AcceptorUseCase;
+import com.palantir.history.models.ImmutableRawLearnerAndAcceptorRecords;
+import com.palantir.history.models.LearnerUseCase;
+import com.palantir.history.models.RawLearnerAndAcceptorRecords;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.NamespaceAndUseCase;
+import com.palantir.paxos.PaxosAcceptorState;
+import com.palantir.paxos.PaxosRound;
+import com.palantir.paxos.PaxosValue;
+import com.palantir.paxos.SqlitePaxosStateLog;
+
+public final class SqlitePaxosStateLogHistory {
+    private final Jdbi jdbi;
+
+    private SqlitePaxosStateLogHistory(Jdbi jdbi) {
+        this.jdbi = jdbi;
+    }
+
+    public static SqlitePaxosStateLogHistory create(DataSource dataSource) {
+        Jdbi jdbi = Jdbi.create(dataSource).installPlugin(new SqlObjectPlugin());
+        jdbi.withExtension(SqlitePaxosStateLog.Queries.class, SqlitePaxosStateLog.Queries::createTable);
+        jdbi.getConfig(JdbiImmutables.class)
+                .registerImmutable(Client.class, PaxosRound.class, NamespaceAndUseCase.class);
+        jdbi.registerRowMapper(new AcceptorPaxosRoundMapper())
+                .registerRowMapper(new LearnerPaxosRoundMapper())
+                .registerRowMapper(new NamespaceAndUseCaseMapper());
+        return new SqlitePaxosStateLogHistory(jdbi);
+    }
+
+    public Set<NamespaceAndUseCase> getAllNamespaceAndUseCaseTuples() {
+        return execute(dao -> dao.getAllNamespaceAndUseCaseTuples());
+    }
+
+    public RawLearnerAndAcceptorRecords getRawLearnerAndAcceptorLogsSince(
+            Client namespace, LearnerUseCase learnerUseCase, AcceptorUseCase acceptorUseCase, long seq) {
+        return execute(dao -> ImmutableRawLearnerAndAcceptorRecords.of(
+                dao.getLearnerLogsSince(namespace, learnerUseCase.value(), seq),
+                dao.getAcceptorLogsSince(namespace, acceptorUseCase.value(), seq)));
+    }
+
+    private <T> T execute(Function<Queries, T> call) {
+        return jdbi.withExtension(Queries.class, call::apply);
+    }
+
+    public interface Queries {
+        @SqlQuery("SELECT DISTINCT namespace, useCase FROM paxosLog")
+        Set<NamespaceAndUseCase> getAllNamespaceAndUseCaseTuples();
+
+        @SqlQuery("SELECT seq, val FROM paxosLog "
+                + "WHERE namespace = :namespace.value AND useCase = :useCase AND seq > :seq "
+                + "ORDER BY seq ASC LIMIT 500")
```

Comment:
How did we chose this limit? We need to figure out the correct limit for remotes

Reply:
you can make it a todo, but this is also fine